main.go

```diff
@@ -34,9 +39,22 @@ func Main() int {
 		FormatOnly: *format,
 		Compact:    *compact,
 	}
+
+	if *cpuprofile != "" {
+		f, err := os.Create(*cpuprofile)
+		if err != nil {
+			log.Fatalf("can't open file for cpu profile: %v", err)
+		}
+		err = pprof.StartCPUProfile(f)
+		if err != nil {
+			log.Fatalf("can't start cpu profile: %v", err)
+		}
+		log.Infof("Writing cpu profile to %s", *cpuprofile)
+		defer pprof.StopCPUProfile()
+	}
 	err := extensions.Init()
 	if err != nil {
-		log.Fatalf("Error initializing extensions: %v", err)
+		return log.FErrf("Error initializing extensions: %v", err)
```

Comment:
Why using your ferry here, and using Fatalf in previous if block you add

Reply:
?

Reply:
ic you mean why not change all of them, only changed what the linter complained about re defer not to running but probably should be all yes, in theory/for consistency

Reply:
done

Reply:
Thanks

Reply:
of note there are still a few more log.Fatalf in other functions (like if you fail to pass a valid filename) and it's fine... for now at least (just means some pprof defer won't happen)